pkg/operator/controller/ingress/status.go

```diff
@@ -246,6 +247,32 @@ func computeIngressAvailableCondition(conditions []operatorv1.OperatorCondition)
 	}
 }
 
+// computeDeploymentProgressingCondition computes the current Progressing status
+// of the related Deployment
+func computeDeploymentProgressingCondition(deployment *appsv1.Deployment) operatorv1.OperatorCondition {
+	var progressingConditions []*appsv1.DeploymentCondition
+
+	for _, dc := range deployment.Status.Conditions {
+		if dc.Type == appsv1.DeploymentProgressing {
+			progressingConditions = append(progressingConditions, &dc)
```

Comment:
That's a good point; originally I intended on surfacing the progressing reason but that felt like that was redundant since you could always check the Deployment itself. I'll simplify this.

Reply:
Did a rewrite of this in the latest push so marking this as resolved for now.